common/tar/platform/linux/archiver.hpp

```diff
@@ -0,0 +1,83 @@
+#ifndef MENDER_ARCHIVER_HPP
+#define MENDER_ARCHIVER_HPP
+
+#include <vector>
+
+#include <archive.h>
+#include <archive_entry.h>
+#include <sys/types.h>
+
+#include <common/io.hpp>
+#include <common/log.hpp>
+#include <common/error.hpp>
+#include <common/expected.hpp>
+
+namespace archiver {
+
+using namespace std;
+
+using mender::common::expected::Expected;
+
+using ExpectedBool = Expected<bool, mender::common::error::Error>;
+
+namespace error = mender::common::error;
+namespace log = mender::common::log;
+
+class ReaderContainer {
+public:
+	mender::common::io::Reader &reader_;
+	std::vector<uint8_t> buff_;
```

Comment:
These are actually accessed in the `reader_callback` from `libarchive`.

So, `private` and `friend`? 
Keep them private, and accessor methods (not my preferred approach here, since they're both accessed)
Change the variables name to remove the `_` suffix

Reply:
If accessors are undesirable (which they usually are unless you want outsiders to also have access to them), then `private` and `friend`, yes.